source/declarations.tex

```diff
@@ -250,21 +256,66 @@
 \grammarterm{explicit-specialization}\iref{temp.expl.spec}.
 \end{note}
 
+\pnum
+If a \grammarterm{static_assert-message}
+matches the syntactic requirements of \grammarterm{unevaluated-string},
+it is an \grammarterm{unevaluated-string} and
+the text of the \grammarterm{static_assert-message} is
+the text of the \grammarterm{unevaluated-string}.
+Otherwise, a \grammarterm{static_assert-message} shall be an expression $M$
+such that
+\begin{itemize}
+\item
+the expression \tcode{$M$.size()} is
+implicitly convertible to the type \tcode{std::size_t}, and
+\item
+the expression \tcode{$M$.data()} is
+implicitly convertible to the type ``pointer to \keyword{const} \keyword{char}''.
+\end{itemize}
+
 \pnum
 \indextext{\idxcode{static_assert}}%
 In a \grammarterm{static_assert-declaration},
-the \grammarterm{constant-expression}
+the \grammarterm{constant-expression} $E$
 is contextually converted to \keyword{bool} and
 the converted expression shall be a constant expression\iref{expr.const}.
-If the value of the expression when
-so converted is \tcode{true}
-or the expression is evaluated in the context of a template definition,
-the declaration has no
-effect. Otherwise,
-the \grammarterm{static_assert-declaration} \defnx{fails}{\idxcode{static_assert}!failed},
-the program is ill-formed, and the resulting
-diagnostic message\iref{intro.compliance} should include the text of
-the \grammarterm{string-literal}, if one is supplied.
+If the value of the expression $E$ when so converted is \tcode{true} or
+the expression is evaluated in the context of a template definition,
+the declaration has no effect and
+the \grammarterm{static_assert-message} is
+an unevaluated operand\iref{expr.context}.
```

Comment:
```suggestion
an unevaluated operand\iref{term.unevaluated.operand}.
```

Reply:
I think we need a check for this.
To avoid not using a label introduced for a term.